src/modules/keyboardmanager/KeyboardManagerEditorLibrary/BufferValidationHelpers.cpp

```diff
@@ -29,53 +29,31 @@ namespace BufferValidationHelpers
         if (selectedKeyCode != -1)
         {
             // Check if the value being set is the same as the other column
-            if (remapBuffer[rowIndex].first[std::abs(int(colIndex) - 1)].index() == 0)
+            const auto otherColumnIndex = static_cast<size_t>(std::abs(colIndex - 1));
+            const auto& otherColumnKey = remapBuffer.at(rowIndex).mapping.at(otherColumnIndex);
+            if (otherColumnKey.index() == 0)
             {
-                DWORD otherColumnKeyCode = std::get<DWORD>(remapBuffer[rowIndex].first[std::abs(int(colIndex) - 1)]);
+                const auto otherColumnKeyCode = std::get<DWORD>(otherColumnKey);
                 if (otherColumnKeyCode == selectedKeyCode || IsKeyRemappingToItsCombinedKey(selectedKeyCode, otherColumnKeyCode))
                 {
                     errorType = ShortcutErrorType::MapToSameKey;
                 }
             }
 
-            // If one column is shortcut and other is key no warning required
```

Comment:
Just removing this check is not enough. I believe we should only allow remapping the same key twice if one remapping is `Alone` and the other one is `In combination`.. This way UI allows you to remap same key twice but in the background it actually doesn't do it. And this is not UI behavior we would want. Let's keep it clear

Reply:
Currently returning an error here blocks further edits on the remap. I deleted this check to allow tentative states with multiple duplicated remaps. Otherwise, the user cannot add a remap on the same key (using another condition to be updated later). If we want to show the error message immediately, I suggest showing it without resetting the combo box.